File: JavaScript/Easy/reverseString.js
// I IMPLEMENT THREE WAYS TO REVERSE STRING FELL FREE TO CHECK BOTH

function reverseString(s)
{
  return s.split("").reverse().join("");
}

// function reverseString(s)
// {
//   let arr = s.split("");
//   for (let i = 0; i < arr.length / 2; i++)
//     [arr[i], arr[arr.length - 1 - i]] = [arr[arr.length - 1 - i], arr[i]];

//   return arr.join("");
// }

// function reverseString(s) {
//   let newS = "";

//   for (let i = s.length - 1; i >= 0; i--)
//     newS += s[i];
//   return newS;
// }

console.log(reverseString("Hello World!"));

File: c++/Easy/reverse-string.cpp
#include <iostream>

// Write a function that reverses a string.

using namespace std;

string reverseString(string s) {
  string revStr = "";

  for (int i = s.length() - 1; i >= 0; i--) {
    revStr += s[i];
  }
  return revStr;
}

int main() {
  cout << reverseString("Hello world!") << endl;
  return 0;
}
